lib/db/client-instance-store.js

```diff
@@ -26,14 +26,25 @@ class ClientInstanceStore {
     constructor(db, getLogger) {
         this.db = db;
         this.logger = getLogger('client-instance-store.js');
+        this.client = this.db.fn.client.config.client;
         const clearer = () => this._removeInstancesOlderThanTwoDays();
         setTimeout(clearer, 10).unref();
         setInterval(clearer, ONE_DAY).unref();
     }
 
     _removeInstancesOlderThanTwoDays() {
+        let whereClause = `BAD_ENGINE=${this.client}`;
```

Comment:
instead of moving this here, it could be better isolated to the provider specific driver. So for example a transformDateExpression function on a mysql adapter, a postgres adapter, a sqlite adapter.